com.stansassets.build/Editor/Steps/UnityPlayerBuildStep.cs

```diff
@@ -0,0 +1,112 @@
+
+        private event Action<BuildStepResultArgs> m_OnCompleteCallback;
+
+        public UnityPlayerBuildStep(List<IBuildTask> tasks)
+        {
+            s_Tasks = tasks;
+        }
+        
+        public void Execute(IBuildStepContext buildContext, Action<BuildStepResultArgs> onComplete = null)
+        {
+            m_OnCompleteCallback = onComplete;
+
+            m_BuildContext = buildContext;
+
+            SetTasksContext();
+            RunTasks();
+                
+            BuildProject();
+        }
+
+        private void BuildProject()
+        {
+            BuildReport report = BuildPipeline.BuildPlayer(m_BuildContext.BuildPlayerOptions);
+
+            BuildSummary summary = report.summary;
+
+            BuildStepResultArgs resultArgs = new BuildStepResultArgs
+            {
+                Step = this,
+                IsSuccess = summary.result == BuildResult.Succeeded,
+                ResultMessage = "UnityPlayerBuildStep execution finish is " + summary.result,
+            };
+
+            m_OnCompleteCallback?.Invoke(resultArgs);
+            m_OnCompleteCallback = null;
+        }
+
+        private void SetTasksContext()
+        {
+            for (int i = 0; i < s_Tasks.Count; i++)
+            {
+                s_Tasks[i].SetContext(m_BuildContext);
+            }
+        }
+        
+        [PostProcessSceneAttribute]
+        private static void RunTasks()
+        {    
+            if(Application.isPlaying)
+                return;
+            
+            Scene currentScene = SceneManager.GetActiveScene();
+            GameObject[] rootGameObjects = currentScene.GetRootGameObjects();
+
+            List<GameObject> currentSceneObjects = GetAllObjects(rootGameObjects);
+
+            for (int i = 0; i < s_Tasks.Count; i++)
+            {
+                for (int j = 0; j < currentSceneObjects.Count; j++)
+                {
+                    List<Component> allObjectComponents = GetAllComponents(currentSceneObjects[j]);
+                    s_Tasks[i].OnPostprocessGameObject(currentSceneObjects[j], allObjectComponents);
+                }
+            }
+        }
+
+        private static List<GameObject> GetAllObjects(GameObject[] rootGameObjects)
+        {
+            List<GameObject> allObjects = new List<GameObject>();
+            
+            for (int i = 0; i < rootGameObjects.Length; i++)
+            {
+                var currentObjectChilds =
+                    rootGameObjects[i].GetComponentsInChildren<Transform>();
+
+                for (int j = 0; j < currentObjectChilds.Length; j++)
+                {
+                    allObjects.Add(currentObjectChilds[j].gameObject);
+                }
+            }
+
+            return allObjects;
+        }
+        
+        private static List<Component> GetAllComponents(GameObject rootObject)
+        {
+            List<Component> allComponents = rootObject.GetComponents(typeof(MonoBehaviour)).ToList();
```

Comment:
And btw overall logic is quite heavy here and we can replace taking all Transforms and then getting Components for each of them with taking all Transforms (to get a list of GOs) and GetComponentsInChildren for each scene root object (we can store it inside DIctionary<GameObject, List<Component>>). Because if we gonna have scene with thousands of GOs - it will take ages to ask Components from each GO.

Reply:
Lets skip optimization for now and make a test later with this please :)